compiler/src/dmd/denum.d

```diff
@@ -169,7 +169,12 @@ extern (C++) final class EnumDeclaration : ScopeDsymbol
             return defaultval;
         }
         //printf("EnumDeclaration::getDefaultValue() %p %s\n", this, toChars());
-        if (defaultval)
+        // https://issues.dlang.org/show_bug.cgi?id=23904
+        // Return defaultval only if it is not ErrorExp.
+        // A speculative context may set defaultval to ErrorExp;
+        // subsequent non-speculative contexts need to be able
+        // to print the error.
+        if (defaultval && !defaultval.isErrorExp())
             return defaultval;
```

Comment:
`ErrorExp.get()` requires that an error has been printed, or it prints "Error: unknown, please file report on issues.dlang.org"

Reply:
And `resolvePropertiesX` does:
```D
            if (e2.op == TOK.error) // because of opaque enum
                return ErrorExp.get(); // requires error message to have been printed
```

Reply:
Yeah, the error handling system is a mess.

Reply:
OK, as long as it doesn't segfault

Reply:
If only there was some kind of way for rolling back exception control flow

Reply:
You can bypass .get if you must btw. The impl just returns a shared global